NitroxClient/Communication/Packets/Processors/ScheduleProcessor.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using NitroxClient.Communication.Packets.Processors.Abstract;
+using NitroxModel.Logger;
+using NitroxModel.Packets;
+using Story;
+
+namespace NitroxClient.Communication.Packets.Processors
+{
+    public class ScheduleProcessor : ClientPacketProcessor<Schedule>
+    {
+        public override void Process(Schedule schedulePacket)
+        {
+            Dictionary<string, PDALog.Entry> entries = (Dictionary<string, PDALog.Entry>)(typeof(PDALog).GetField("entries", BindingFlags.NonPublic | BindingFlags.Static).GetValue(null));
+            ScheduledGoal goal = new ScheduledGoal();
+            goal.version = 1;
+            goal.goalKey = schedulePacket.Key;
+            goal.goalType = (Story.GoalType)System.Enum.Parse(typeof(Story.GoalType), schedulePacket.Type);
+            goal.timeExecute = schedulePacket.TimeExecute;
+            if (goal.timeExecute >= DayNightCycle.main.timePassedAsDouble)
+            {
+                if (!StoryGoalScheduler.main.schedule.Any(alreadyInGoal => alreadyInGoal.goalKey == goal.goalKey) && !entries.TryGetValue(goal.goalKey, out PDALog.Entry value) && !StoryGoalManager.main.completedGoals.Contains(schedulePacket.Key))
+                {
+                    StoryGoalScheduler.main.schedule.Add(goal);
+                }
+            }
+            Log.Debug($"Processed a Schedule packet [{goal}]");
+        }
+    }
```

Comment:
This code can be simplified in terms of readability (btw: verify if my assumptions - and comments - are correct)

```suggestion
        private readonly FieldInfo entriesField = typeof(PDALog).GetField("entries", BindingFlags.NonPublic | BindingFlags.Static);

        public override void Process(Schedule schedulePacket)
        {
            ScheduledGoal goal = new()
            {
                version = 1,
                goalKey = schedulePacket.Key,
                goalType = (Story.GoalType)Enum.Parse(typeof(Story.GoalType), schedulePacket.Type),
                timeExecute = schedulePacket.TimeExecute
            };

            if (ShouldSchedule(goal.timeExecute) && !IsAlreadyKnown(goal.goalKey))
            {
                StoryGoalScheduler.main.schedule.Add(goal);
            }
            Log.Debug($"Processed a Schedule packet [{goal}]");
        }

        private bool ShouldSchedule(float timeToExecute)
        {
            return timeToExecute >= DayNightCycle.main.timePassedAsDouble;
        }

        private bool IsAlreadyKnown(string goalKey)
        {
            Dictionary<string, PDALog.Entry> entries = (Dictionary<string, PDALog.Entry>)entriesField.GetValue(null);

            return StoryGoalScheduler.main.schedule.All(g => g.goalKey == goalKey) //  Scheduled
                   || entries.TryGetValue(goalKey, out PDALog.Entry _) //  Registered
                   || StoryGoalManager.main.completedGoals.Contains(goalKey); // Completed
        }
```